aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -830,6 +830,83 @@ For more information on JavaScript isolation with JavaScript modules, see <xref:
 
 :::moniker-end
 
+:::moniker range=">= aspnetcore-8.0"
+
+## Hashed routing to named elements
+
+Navigate to a hashed (`#`) named element using the following approaches:
+
+* Anchor element (`<a>`) with an `href`.
+* <xref:Microsoft.AspNetCore.Components.Routing.NavLink> component with an `href`.
+* <xref:Microsoft.AspNetCore.Components.NavigationManager.NavigateTo%2A?displayProperty=nameWithType> passing the URL.
+
+The following example demonstrates hashed routing to named H2 headings.
+
+In the `Index` (`Pages/Index.razor`) and `Counter` (`Pages/Counter.razor`) components, place the following markup at the bottom of the existing component markup:
+
+```razor
+<div class="border border-info rounded bg-info" style="height:500px"></div>
+
+<h2 id="targetElement">Target H2 heading</h2>
+<p>Content!</p>
+```
+
+Add the following `HashedRouting` component to the app:
+
+`Pages/HashedRouting.razor`:
+
+```razor
+@page "/hashed-routing"
+@inject NavigationManager Navigation
+
+<PageTitle>Hashed routing</PageTitle>
+
+<h1>Hashed routing to named elements</h1>
+
+<ul>
+    <li>
+        <a href="#targetElement">
+            Anchor in this component
+        </a>
+    </li>
+    <li>
+        <a href="/#targetElement">
+            Anchor to the <code>Index</code> component
+        </a>
+    </li>
+    <li>
+        <a href="/counter#targetElement">
+            Anchor to the <code>Counter</code> component
+        </a>
+    </li>
+    <li>
+        <NavLink href="#targetElement">
+            Use a `NavLink` component in this component
+        </NavLink>
+    </li>
+    <li>
+        <button @onclick="NavigateToElement">
+            Navigate with <code>NavigationManager</code> to the 
+            <code>Counter</code> component
+        </button>
+    </li>
+</ul>
+
+<div class="border border-info rounded bg-info" style="height:500px"></div>
+
+<h2 id="targetElement">Target H2 heading</h2>
```

Comment:
For some elements in the example let's use name instead of id to show that we support scrolling to the elements with id attribute and name attribute.

Reply:
Ok ... I'm *BACK!* 😄 

Named elements (`name` attribute) are depreciated under HTML5, which I'm certain that you know, and Blazor adopts the current standard HTML5 by doctype by default, which I'm also certain that you know. Therefore, mentioning this we ***should*** tell devs such is the case. I don't think we want to support their use in HTML5 ... or show it for that matter in examples. I admit two things about this ...

* Over the years as a professional dev, I really disliked it when orgs threw standards out the window. I swore a 🩸 *blood oath* 🩸 to the 🙏 *server gods* 🙏 that if I can ever stop organizations from doing it that I would.
* I ***have no power here*** 😆 to stop MS from publishing standards-breaking text or code.

Here's my recommendation if you want to mention it. I suggest in a NOTE to the effect ...

> \> \[!NOTE]
> \> Named elements for targeting anchor elements with the \`name\` attribute are \[depreciated as of HTML5](https://developer.mozilla.org/docs/Web/HTML/Element/a#deprecated_attributes), and Blazor adopts the current HTML specification by \[Doctype](https://developer.mozilla.org/docs/Glossary/Doctype) in \`wwwroot/index.html\` (Blazor WebAssembly) and \`Pages/_Host.cshtml\` (Blazor Server).
> \>
> \> Blazor scrolls to named targets in the same fashion that it scrolls to elements by \`id\`. However, we recommend using \`id\` to target elements in Blazor based on the current HTML specification.

Otherwise tho if you insist, please send code suggestion(s) for the lines that you want to update. I'm sure that I'll grind my teeth over it ... and my dentist 🦷👨‍⚕️ won't be pleased ... *but I'll live!* 🤣

Reply:
I actually didn't know that `name` attribute was deprecated. Thanks! In that case I may change our code to support only `id`. If the code will stay the same let's use the note to prefer id over name as you suggested. I'll give the final answer tomorrow.

Reply:
@guardrex, let's not provide example with name. I will remove support for name attribute.